src/components/fields/contact-number/contact-number.tsx

```diff
@@ -0,0 +1,196 @@
+					(value) => {
+						if (!value || !singaporeRule) return true;
+
+						const validationRules = singaporeRule.singaporeNumber;
+						const hasAdditionalValidations = isObject(validationRules);
+
+						if (!hasAdditionalValidations && singaporeRule.singaporeNumber) {
+							return PhoneHelper.isSingaporeNumber(value, true) || PhoneHelper.isSingaporeNumber(value);
+						}
+
+						if (hasAdditionalValidations && validationRules?.homeNumber) {
+							return PhoneHelper.isSingaporeNumber(value, true);
+						}
+
+						if (hasAdditionalValidations && validationRules?.mobileNumber) {
+							return PhoneHelper.isSingaporeNumber(value);
+						}
+					}
+				)
+				.test(
+					"internationalNumber",
+					internationalRule?.errorMessage || ERROR_MESSAGES.CONTACT.INVALID_INTERNATIONAL_NUMBER,
+					(value) => {
+						if (!value || !internationalRule) return true;
+
+						return PhoneHelper.isInternationalNumber(selectedCountry?.name, value);
+					}
+				),
+			validation
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation, selectedCountry]);
+
+	useEffect(() => {
+		setStateValue(value || "");
+	}, [value]);
+
+	useEffect(() => {
+		const countryName = country || "Singapore";
+
+		setSelectedCountry({
+			name: countryName,
+			prefix: getPrefix(countryName),
+		});
+	}, [country]);
+
+	// =============================================================================
+	// EVENT HANDLERS
+	// =============================================================================
+	const handlePrefixChange = (key: TCountry, prefix: string): void => {
+		const { number: phoneNumberWithoutPrefix } = PhoneHelper.getParsedPhoneNumber(stateValue);
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(prefix, phoneNumberWithoutPrefix);
+
+		setSelectedCountry({ name: key, prefix });
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+		const phoneNumberWithPrefix = PhoneHelper.formatPhoneNumber(selectedCountry?.prefix, event.target.value);
+
+		onChange({ target: { value: phoneNumberWithPrefix } });
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const formatDisplayValue = (): string => {
+		const parsedValues = stateValue.split(" ");
+		return parsedValues.length > 1 ? parsedValues[1] : stateValue;
+	};
+
+	const getSpacing = (): number => {
+		if (selectedCountry?.name === "Singapore") return 4;
+
+		return 0;
+	};
+
+	const getMaxLength = (): number => {
+		if (selectedCountry?.name === "Singapore") {
+			return 9;
+		}
+		return 14;
+	};
+
+	const getAddOns = (): AddonProps<string, unknown> => {
+		const singaporeRuleValidation = singaporeRule?.singaporeNumber;
+
+		let isInternationalNumbers = true;
+
+		if (singaporeRuleValidation) {
```

Comment:
my apologies, it seems like the country format is getting complicated, I see by design we want to keep singapore number validation and international number validation mutually exclusive but here it is possible to set both validation true.

instead of having two rules, can we have one instead so devs are forced to pick either singapore or international validation?

along the same line, we wanna reduce confusion with the home/mobile number validation, it is only applicable to singapore numbers and devs can turn on one or both validations, can consider using one key instead.

```js
// current
{
  homeNumber: true,
  mobileNumber: true
}

// new?
<name of key>: "home"|"mobile"|"both"
```